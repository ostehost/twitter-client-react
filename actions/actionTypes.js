const actionTypes = {
    CHANGE_SEARCH: 'CHANGE_SEARCH',
    CHANGE_COUNT: 'CHANGE_COUNT',
    CHANGE_TYPE: 'CHANGE_TYPE',
    LOAD_TWEETS: 'LOAD_TWEETS',
    TWEETS_LOADED: 'TWEETS_LOADED',
    LOAD_TWEETS_FAILURE: 'LOAD_TWEETS_FAILURE',
    SORT: 'SORT'
}

export default actionTypes;