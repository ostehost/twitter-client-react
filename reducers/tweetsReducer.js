import sortItems from '../lib/sortItems'

import actionTypes from '../actions/actionTypes';

export const initialState = {
    search: '',
    count: '10',
    type: false,
    tweets: [],
    tweets_loading: false,
    sort_by: 'created_at',
    sort_order: 1,
    error: null
}

const tweetsReducer = (state = initialState, action) => {
    switch (action.type) {
        case actionTypes.CHANGE_TYPE:
            return { ...state, type: action.value };
        case actionTypes.CHANGE_COUNT:
            return { ...state, count: action.value };
        case actionTypes.CHANGE_SEARCH:
            return { ...state, search: action.value };
        case actionTypes.LOAD_TWEETS:
            return { ...state, tweets: [], tweets_loading: true };
        case actionTypes.TWEETS_LOADED:
            let tweets = action.append ? [action.payload, ...state.tweets] : action.payload;
            sortItems(tweets, state.sort_by, state.sort_order);
            return { ...state, error: null, tweets: tweets, tweets_loading: false };
        case actionTypes.SORT:
            sortItems(state.tweets, action.sortBy, action.sortOrder);
            return { ...state, sort_by: action.sortBy, sort_order: action.sortOrder }
        case actionTypes.LOAD_TWEETS_FAILURE:
            return {
                ...state,
                error: action.error,
                tweets_loading: false
            }
        default:
            return state
    }
}

export default tweetsReducer;