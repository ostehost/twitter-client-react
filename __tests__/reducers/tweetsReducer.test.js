// Reducer to be tested
import tweetsReducer from '../../reducers/tweetsReducer';
import actionTypes from '../../actions/actionTypes';

describe('tweetsReducer', () => {

    test('is correct', () => {
        const action = { type: null };
        expect(tweetsReducer(undefined, action)).toMatchSnapshot();
    });

    describe('CHANGE_TYPE', () => {
        test('is correct', () => {
            const action = { type: actionTypes.CHANGE_TYPE, value: "test" };
            // const initialState = { search: "test" };
            // expect(tweetsReducer(undefined, action)).toEqual(initialState);
            expect(tweetsReducer(undefined, action)).toMatchSnapshot();
        });
    });

    describe('CHANGE_COUNT', () => {
        test('is correct', () => {
            const action = { type: actionTypes.CHANGE_COUNT, value: "test" };
            expect(tweetsReducer(undefined, action)).toMatchSnapshot();
        });
    });

    describe('CHANGE_SEARCH', () => {
        test('is correct', () => {
            const action = { type: actionTypes.CHANGE_SEARCH, value: "test" };
            expect(tweetsReducer(undefined, action)).toMatchSnapshot();
        });
    });

    describe('LOAD_TWEETS', () => {
        test('is correct', () => {
            const action = { type: actionTypes.LOAD_TWEETS };
            expect(tweetsReducer(undefined, action)).toMatchSnapshot();
        });
    });

    describe('TWEETS_LOADED, ', () => {
        test('is correct', () => {
            const action = { type: actionTypes.TWEETS_LOADED, payload: "payload", append: true};
            expect(tweetsReducer({tweets: []}, action)).toMatchSnapshot();
        });
    });

    describe('SORT, ', () => {
        test('is correct', () => {
            const action = { type: actionTypes.SORT, sortBy: "created_at", sortOrder: 1 };
            expect(tweetsReducer({tweets: []}, action)).toMatchSnapshot();
        });
        test('is correct', () => {
            const action = { type: actionTypes.SORT, sortBy: "created_at", sortOrder: -1 };
            expect(tweetsReducer({tweets: []}, action)).toMatchSnapshot();
        });
    });

    describe('LOAD_TWEETS_FAILURE, ', () => {
        test('is correct', () => {
            const action = { type: actionTypes.LOAD_TWEETS_FAILURE, error: "error" };
            expect(tweetsReducer(undefined, action)).toMatchSnapshot();
        });
    });
});