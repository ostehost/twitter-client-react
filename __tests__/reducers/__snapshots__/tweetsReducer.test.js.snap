// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tweetsReducer CHANGE_COUNT is correct 1`] = `
Object {
  "count": "test",
  "error": null,
  "search": "",
  "sort_by": "created_at",
  "sort_order": 1,
  "tweets": Array [],
  "tweets_loading": false,
  "type": false,
}
`;

exports[`tweetsReducer CHANGE_SEARCH is correct 1`] = `
Object {
  "count": "10",
  "error": null,
  "search": "test",
  "sort_by": "created_at",
  "sort_order": 1,
  "tweets": Array [],
  "tweets_loading": false,
  "type": false,
}
`;

exports[`tweetsReducer CHANGE_TYPE is correct 1`] = `
Object {
  "count": "10",
  "error": null,
  "search": "",
  "sort_by": "created_at",
  "sort_order": 1,
  "tweets": Array [],
  "tweets_loading": false,
  "type": "test",
}
`;

exports[`tweetsReducer LOAD_TWEETS is correct 1`] = `
Object {
  "count": "10",
  "error": null,
  "search": "",
  "sort_by": "created_at",
  "sort_order": 1,
  "tweets": Array [],
  "tweets_loading": true,
  "type": false,
}
`;

exports[`tweetsReducer LOAD_TWEETS_FAILURE,  is correct 1`] = `
Object {
  "count": "10",
  "error": "error",
  "search": "",
  "sort_by": "created_at",
  "sort_order": 1,
  "tweets": Array [],
  "tweets_loading": false,
  "type": false,
}
`;

exports[`tweetsReducer SORT,  is correct 1`] = `
Object {
  "sort_by": "created_at",
  "sort_order": 1,
  "tweets": Array [],
}
`;

exports[`tweetsReducer SORT,  is correct 2`] = `
Object {
  "sort_by": "created_at",
  "sort_order": -1,
  "tweets": Array [],
}
`;

exports[`tweetsReducer TWEETS_LOADED,  is correct 1`] = `
Object {
  "error": null,
  "tweets": Array [
    "payload",
  ],
  "tweets_loading": false,
}
`;

exports[`tweetsReducer is correct 1`] = `
Object {
  "count": "10",
  "error": null,
  "search": "",
  "sort_by": "created_at",
  "sort_order": 1,
  "tweets": Array [],
  "tweets_loading": false,
  "type": false,
}
`;
